openvdb/openvdb/CMakeLists.txt

```diff
@@ -443,6 +469,52 @@ set(OPENVDB_LIBRARY_THREAD_INCLUDE_FILES
   thread/Threading.h
 )
 
+##########################################################################
+
+# Configure auto-generation of explicit template definition translation units
+
+if(USE_EXPLICIT_INSTANTIATION)
+  # Define the list of headers for which to generate new translation units
+  set(OPENVDB_LIBRARY_EXPLICIT_INSTANTIATION_FILES
+    tools/Activate.h
+  )
+
+  # Create an "instantations" dir to host all the auto-generated source files
+  file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/instantiations)
+
+  # A new source file is generated by CMake for every file and tree - for example:
+  # ActivateFloatTree.cc, ActivateDoubleTree.cc, ChangeBackgroundFloatTree.cc
+  foreach(INSTANTIATION_FILE ${OPENVDB_LIBRARY_EXPLICIT_INSTANTIATION_FILES})
+    foreach(INSTANTIATION_TREE ${OPENVDB_LIBRARY_EXPLICIT_INSTANTIATION_TREES})
+
+      # String replace to generate the name of the new source file
+      # "tools/Activate.h" => "ActivateFloatTree.cc"
+      string(REGEX REPLACE "^.*/(.*).h" "\\1"
+        INSTANTIATION_SOURCE "${INSTANTIATION_FILE}${INSTANTIATION_TREE}.cc"
+      )
+
+      # Prefix the source file with the CMake binary dir and the "instantiations" subdir
+      set(INSTANTIATION_CMAKE_SOURCE
+        ${CMAKE_CURRENT_BINARY_DIR}/instantiations/${INSTANTIATION_SOURCE}
+      )
+
+      # The auto-generated file defines the OPENVDB_TREE_TEMPLATE macro for a single tree type
+      # and then the header file itself which defines all functions to be explicitly instantiated.
+      execute_process(
+        COMMAND echo
+          "#define OPENVDB_TREE_TEMPLATE(Function)  template OPENVDB_TEMPLATE_API Function(${INSTANTIATION_TREE});\n"
+          "#include <openvdb/${INSTANTIATION_FILE}>"
+        OUTPUT_FILE ${INSTANTIATION_CMAKE_SOURCE}
+      )
```

Comment:
You might even be able to replace this with `file(WRITE)`:
https://cmake.org/cmake/help/latest/command/file.html#write

Reply:
Yes, that's better, thanks. I made that change now.